workflows/phaseimpute/main.nf

```diff
@@ -218,35 +239,44 @@ workflow PHASEIMPUTE {
 
             }
             if (params.tools.split(',').contains("glimpse2")) {
-                error "Glimpse2 not yet implemented"
 
                 // Use previous chunks if --step panelprep
-                // if (params.panel && params.step.split(',').contains("panelprep") && !params.chunks) {
-                //     ch_chunks = VCF_CHUNK_GLIMPSE.out.chunks_glimpse1
+                if (params.panel && params.step.split(',').find { it in ["all", "panelprep"] } && !params.chunks) {
+                    ch_chunks = VCF_CHUNK_GLIMPSE.out.chunks_glimpse1
+                } else if (params.chunks) {
+                    ch_chunks = ch_chunks.map { chr, txt -> [chr, file(txt)]}
+                            .splitCsv(header: ['ID', 'Chr', 'RegionIn', 'RegionOut', 'Size1', 'Size2'], sep: "\t", skip: 0)
+                            .map { meta, it -> [meta, it["RegionIn"], it["RegionOut"]]}
+                // Use channel ch_chunks for GLIMPSE2 imputation
+                } else {
+                    error "Either no reference panel was included or you did not set step --panelprep or you did not provide --chunks"
+                }
 
-                //     VCF_IMPUTE_GLIMPSE2(ch_input_impute,
-                //                     ch_panel_phased,
-                //                     ch_chunks,
-                //                     ch_fasta)
-                // } else if (params.chunks) {
-                //     // use provided chunks
-                // } else {
-                //     error "Either no reference panel was included or you did not set step --panelprep or you did not provide --chunks"
-                // }
+                // Run imputation
+                VCF_IMPUTE_GLIMPSE2(ch_input_impute,
+                        ch_panel_phased,
+                        ch_chunks,
+                        ch_fasta)
+                ch_versions = ch_versions.mix(VCF_IMPUTE_GLIMPSE2.out.versions)
 
+                // Concatenate by chromosomes
+                CONCAT_GLIMPSE2(VCF_IMPUTE_GLIMPSE2.out.vcf_tbi)
+                ch_versions = ch_versions.mix(CONCAT_GLIMPSE2.out.versions)
 
+                // Add results to input validate
+                ch_input_validate = ch_input_validate.mix(CONCAT_GLIMPSE2.out.vcf_tbi_join)
             }
 
             if (params.tools.split(',').contains("stitch")) {
                 print("Impute with STITCH")
 
                 // Obtain the user's posfile if provided or calculate it from ref panel file
-                if (params.posfile ) {  // User supplied posfile
-                    ch_posfile = ch_posfile
-                } else if (params.panel && params.step.split(',').contains("panelprep")) { // Panelprep posfile
+                if (params.posfile) {  // User supplied posfile
+                    ch_posfile  = ch_posfile
```

Comment:
This line is not necessary. No ?

Reply:
No, because the default will always be the user-supplied `posfile`. I will change this.